youtube_dl/extractor/iqiyi.py

```diff
@@ -301,10 +302,14 @@ def get_raw_data(self, tvid, video_id):
     def _extract_playlist(self, webpage):
         PAGE_SIZE = 50
 
-        links = re.findall(
-            r'<a[^>]+class="site-piclist_pic_link"[^>]+href="(http://www\.iqiyi\.com/.+\.html)"',
-            webpage)
-        if not links:
+        links = []
+        for link in re.findall(r'<a[^>]+class="[^"]*site-piclist_pic_link[^"]*"[^>]*>', webpage):
```

Comment:
This is my idea:

1. get_element_by_id is not touched (by HTML standards IDs are unique)
2. get_element_by_attribute => get_elements_by_attribute
3. get_element_by_class => get_elements_by_class

And replace all usages in extractors. It's better to open a new pull request for this change.